airflow/docker.go

```diff
@@ -568,6 +574,58 @@ func (d *DockerCompose) getWebServerContainerID() (string, error) {
 	return "", err
 }
 
+func (d *DockerCompose) RunTest(dagID, settingsFile, startDate string, noCache, taskLogs bool) error {
+	// Get project containers
+	psInfo, err := d.composeService.Ps(context.Background(), d.projectName, api.PsOptions{
+		All: true,
+	})
+	if err != nil {
+		return errors.Wrap(err, composeCreateErrMsg)
+	}
+	if len(psInfo) > 0 {
+		// Ensure project is not already running
+		for i := range psInfo {
+			if checkServiceState(psInfo[i].State, dockerStateUp) {
+				if strings.Contains(psInfo[i].Name, SchedulerDockerContainerName) {
+					err = d.imageHandler.RunTest(dagID, d.envFile, settingsFile, startDate, psInfo[i].Name, taskLogs)
+					if err != nil {
+						return err
+					}
+					return nil
+				}
+			}
+		}
+	}
+
+	fmt.Println("Building image... For a faster 'astro run' experince run this command while Airflow is running with 'astro dev start'\n ")
+	// add astro-run-dag
+	err = fileutil.AddLineToFile("./requirements.txt", "astro-run-dag", "# This package is needed for the astro run command. It will be removed before a deploy")
+	if err != nil {
+		fmt.Printf("Adding 'astro-run-dag' package to requirements.txt unsuccessful: %s\nManually add package to requirements.txt", err.Error())
+	}
+	// add airflow db init
+	err = fileutil.AddLineToFile("./Dockerfile", "RUN airflow db init", "")
```

Comment:
Is there a way to avoid adding/removing lines to the dockerfile all together? Instead is it possible to run the command directly after logging into the container or via an env var?

Reply:
with this method it gets cached so there is some upside. Id like a way to run it without adding to the Dockerfile. Right now I am just running `docker run <python command>` because this gets the fastest results. If we can find another way to do it that doesn't slow the command down without adding a line that would be great. I think this is an improvement we should plan to add on later